valgrind.supp

```diff
@@ -0,0 +1,965 @@
+   fun:_ZN2ml5maths14testForPeriodsERKNS0_33CPeriodicityHypothesisTestsConfigEllRKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS4_9allocatorISA_EEEE
+   fun:_ZN2ml5maths30CTimeSeriesDecompositionDetail16CPeriodicityTest4testERKNS1_9SAddValueE
+   fun:_ZN2ml5maths30CTimeSeriesDecompositionDetail16CPeriodicityTest6handleERKNS1_9SAddValueE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Cond
+   fun:_ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_
+   fun:_ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m
+   fun:_ZNSt3__1lsIcNS_11char_traitsIcEENS_9allocatorIcEEEERNS_13basic_ostreamIT_T0_EES9_RKNS_12basic_stringIS6_S7_T1_EE
+   fun:_ZNSt3__1lsIdcNS_11char_traitsIcEEEERNS_13basic_ostreamIT0_T1_EES7_RKNS_7complexIT_EE
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE6print_INSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_S5_
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE5printINSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core14printer_detail12CNodePrinterIN5boost17integral_constantIbLb0EEENS0_17CContainerPrinterEE5printINSt3__17complexIdEEEENS9_12basic_stringIcNS9_11char_traitsIcEENS9_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core17CContainerPrinter12printElementINSt3__17complexIdEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__111__wrap_iterIPKNS3_7complexIdEEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEET_SG_
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__16vectorINS3_7complexIdEENS3_9allocatorIS6_EEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS7_IcEEEERKT_
+   fun:_ZN2ml5maths7CSignal3fftERNSt3__16vectorINS2_7complexIdEENS2_9allocatorIS5_EEEE
+   fun:_ZN2ml5maths7CSignal16autocorrelationsERKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEERNS3_IdNS9_IdEEEE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Cond
+   fun:_ZN4dyld5_mainEPK12macho_headermiPPKcS5_S5_Pm
+   fun:_ZN13dyldbootstrap5startEPK12macho_headeriPPKclS2_Pm
+   fun:_dyld_start
+   obj:*
+   obj:*
+   obj:*
+   obj:*
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Cond
+   fun:_ZNK16ImageLoaderMachO9getRPathsERKN11ImageLoader11LinkContextERNSt3__16vectorIPKcNS4_9allocatorIS7_EEEE
+   fun:_ZN11ImageLoader22recursiveLoadLibrariesERKNS_11LinkContextEbRKNS_10RPathChainEPKc
+   fun:_ZN11ImageLoader4linkERKNS_11LinkContextEbbbRKNS_10RPathChainEPKc
+   fun:_ZN4dyld4linkEP11ImageLoaderbbRKNS0_10RPathChainEj
+   fun:_ZN4dyld5_mainEPK12macho_headermiPPKcS5_S5_Pm
+   fun:_ZN13dyldbootstrap5startEPK12macho_headeriPPKclS2_Pm
```

Comment:
`_ZN4dyld4linkEP11ImageLoaderbbRKNS0_10RPathChainEj` runs before `main`, so any problems are out of our control and all the stack trace that include this can be replaced with just one like:

```
  ...
  fun:_ZN4dyld4linkEP11ImageLoaderbbRKNS0_10RPathChainEj
```